### PR TITLE
update contribution manifesto for 4.x convergence onto api.ci prow

### PR DESCRIPTION
**Pull request by gabemontero**
@pradeepto @sthaha @waveywaves - I've updated the contribution guide to detail our PR, CI, and release building processes as they have evolved during the 4.0 time frame onto openshift's prow based system

we'll be reviewing this during the face to face, but please review
a) to develop any productive questioning / answering before the face to face meeting
b) based on how your absorbtion and amount of questions go, I can record a video walking through this prior to the face to face if that will help
c) and of course review the content if I need to make sections more clean

@openshift/sig-developer-experience @bparees @pweil- fyi

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,41 +41,41 @@ Choose either the CentOS7 or RHEL7 based image:
...
+Also note, as of 3.11, the RHEL images are hosted at registry.redhat.io as well.  This is the terms based 
+registry and requires credentials for access.  See [Transitioning the Red Hat container registry](https://www.redhat.com/en/blog/transitioning-red-hat-container-registry) for details:
     * registry.redhat.io/openshift3/jenkins-2-rhel7:v3.11
     * registry.redhat.io/openshift3/jenkins-agent-nodejs-8-rhel7:v3.11
     * registry.redhat.io/openshift3/jenkins-agent-maven-35-rhel7:v3.11
     * registry.redhat.io/openshift3/jenkins-slave-base-rhel7:v3.11
     
-    The openshift cluster install for 3.11 will insure that credentials are provided and subsequently available on the nodes
-    in the cluster to facilitate image pulling.
+The openshift cluster install for 3.11 will insure that credentials are provided and subsequently available on the nodes
```
Comment (adambkaplan) [inline README.md]: /s/insure/ensure/

File: README.md [hunk 2]
```diff
@@ -101,34 +101,31 @@ For example:
 Installation (OpenShift V4)
 ---------------------------------
 
-    Starting with v4.0, the images are no longer hosted on DockerHub, but instead on Quay, for users without entitlements/subscriptions for support.  Their pull specs are:
-    * quay.io/openshift/origin-jenkins:v4.0
-    * quay.io/openshift/origin-jenkins-agent-nodejs:v4.0
-    * quay.io/openshift/origin-jenkins-agent-maven:v4.0
-    * quay.io/openshift/origin-jenkins-agent-base:v4.0
+Starting with v4.0, the images are no longer hosted on DockerHub, but instead on Quay, for users without entitlements/subscriptions for support.  Their pull specs are:
```
Comment (adambkaplan) [inline README.md]:
```suggestion
Starting with v4.0, the images are only available on quay.io for public community support.  Their pull specs are:
```

**Comment by gabemontero** [comment]
updates pushed @adambkaplan

File: README.md [hunk 3]
```diff
@@ -101,34 +101,31 @@ For example:
...
-    more clearly denote that they are for builds on developers' local machines that most likely do not have a Red Hat subscription / entitlement.  The 
-    `Dockerfile.localdev` variants are structured to allow building of the images on machines without `RHEL` subscriptions, even though there base images are no longer based on `CentOS`.  And subscriptions
-    are still required for use of `Dockerfile.rhel7`.
+The `Dockerfile.rhel7` variants still exists, but as part of the `CentOS` vs. `RHEL` distinction no longer existing, the various `Dockerfile` files have been renamed to `Dockerfile.localdev` to more clearly denote that they are for builds on developers' local machines that most likely do not have a Red Hat subscription / entitlement.  The `Dockerfile.localdev` variants are structured to allow building of the images on machines without `RHEL` subscriptions, even though there base images are no longer based on `CentOS`.  And subscriptions are still required for use of `Dockerfile.rhel7`.
```
Comment (grdryn) [inline README.md]:
```suggestion
The `Dockerfile.rhel7` variants still exists, but as part of the `CentOS` vs. `RHEL` distinction no longer existing, the various `Dockerfile` files have been renamed to `Dockerfile.localdev` to more clearly denote that they are for builds on developers' local machines that most likely do not have a Red Hat subscription / entitlement.  The `Dockerfile.localdev` variants are structured to allow building of the images on machines without `RHEL` subscriptions, even though the base images are no longer based on `CentOS`.  Subscriptions are still required building with `Dockerfile.rhel7`.
```

File: CONTRIBUTING_TO_OPENSHIFT_JENKINS_IMAGE_AND_PLUGINS.md [hunk 4]
```diff
@@ -0,0 +1,531 @@
...
+
+You are now ready for the next step
+
+### Step 2: updating OSBS/Brew for generating the officially supported images available with Red Hat subscriptions
+
+First, some background:  for all Red Hat officially supported content, to ensure protection from outside hackers, all content is built in a quarantined system with no access to the external Internet.  As such, we have to inject all content into OpenShift's Brew server (see links like [https://docs.brew.sh/Homebrew-on-Linux](https://docs.brew.sh/Homebrew-on-Linux) if you are interested in the details/histories of this Linux build infrastructure), which is then scrubbed before official builds are run with it.  The injection is specifically the creation of an RPM which contains all the plugin binaries.
```
Comment (grdryn) [inline CONTRIBUTING_TO_OPENSHIFT_JENKINS_IMAGE_AND_PLUGINS.md]: AIUI the "Brew" build server here doesn't have anything to with "homebrew", which is a package manager for MacOS (and, according to the link here, also Linux).

"Brew" in the case of this build server is instead just the name of an instance of "Koji", so if there's going to be a link here, I'd suggest these two (Koji and OSBS):

https://pagure.io/koji/
https://osbs.readthedocs.io/en/latest/

File: README.md (hunk 3 above)
Comment (gabemontero) [reply README.md]: part of the next push - thanks @grdryn

File: CONTRIBUTING_TO_OPENSHIFT_JENKINS_IMAGE_AND_PLUGINS.md (hunk 4 above)
Comment (gabemontero) [reply CONTRIBUTING_TO_OPENSHIFT_JENKINS_IMAGE_AND_PLUGINS.md]: cool ! ... thanks for the education @grdryn - updates will be part of the next push

**Comment by gabemontero** [comment]
updates pushed @grdryn

**Comment by gabemontero** [comment]
thanks for the review @adambkaplan @grdryn 

@pradeepto @sthaha @waveywaves - it occurred to me that it might be easier for you folks to read/consume this in acutal webpage / rendered markdown form :-)

So I'm going to go ahead and merge this.  If, as you folk read this, you uncover more clarifications, etc. that are needed, we can track via separate issues/emails/PRs as needed.

**Comment by gabemontero** [comment]
also @pradeepto @sthaha @waveywaves - this would be a good opportunity for you to examine the output from the ci/prow jobs from *this* PR and correlate them with the updates to the doc here :-)

**Comment by gabemontero** [comment]
api.ci mirros got busted again last night ... reached out to art but no response yet

**Pull request merged**